src/components/GenericVectorBaseMap/GenericVectorBaseMap.js

```diff
@@ -0,0 +1,160 @@
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { MapContainer, useMap } from 'react-leaflet';
+import 'leaflet/dist/leaflet.css';
+import 'leaflet.vectorgrid';
+import L from 'leaflet';
+import 'proj4';
+import 'proj4leaflet';
+import { projCRSOptions } from '../../utils/crs';
+
+const LabelsLayer = ({ wmsUrl, wmsOptions }) => {
```

Comment:
My intention when creating a separate component for the label layer was to render the labels independently of the feature layer. This way, if one layer is rendered faster than the other it would give users a higher perceived performance by presenting some information in a tile space. This got me thinking of the layer independence and I've added a zIndex option so that labels will render on top of the climate rasters, making them even more readable.

Reply:
I get the reason for having a separate label layer and feature layer. That's excellent.

But the label layer is just a WMS layer, and that is already built in to Leaflet. I think you can just use that, passing in the appropriate parameters to make it fetch the label tiles. What am I missing here?

(Possibly you are not distinguishing between a _component_ and _what it renders_, which is parallel to the distinction between a class and an instance. I'm saying we don't need a new component, but we do need of course a separate instance of it for the label layer.)